example/example.client.ts

```diff
@@ -210,21 +204,49 @@ export type Implementation = (
 
 export class ExpressZodAPIClient {
   constructor(protected readonly implementation: Implementation) {}
-  public readonly provide: Provider = async (method, path, params) =>
-    this.implementation(
+  /** @deprecated use the overload with 2 arguments instead */
+  public provide<M extends Method, P extends Path>(
+    method: M,
+    path: P,
+    params: `${M} ${P}` extends keyof Input
+      ? Input[`${M} ${P}`]
+      : Record<string, any>,
+  ): Promise<
+    `${M} ${P}` extends keyof Response ? Response[`${M} ${P}`] : unknown
+  >;
+  public provide<K extends MethodPath>(
+    request: K,
+    params: Input[K],
+  ): Promise<Response[K]>;
+  public provide(
+    ...args:
+      | [string, string, Record<string, any>]
+      | [string, Record<string, any>]
+  ) {
+    const [method, path, params] = (
+      args.length === 2 ? [...args[0].split(" "), args[1]] : args
```

Comment:
I'm concerned about this `.split(" ")`.
I suppose it's possible for a path to contain spaces

Reply:
addressed in 68e0ff04